### ISSUE TITLE
What do I need to take care using BatchNorm?

### ISSUE THREAD

**Issue by datistiquo** [keras-team/keras#13419]
Batchnorm changed  a lot and reading here and on stackoverflow somehow confuses, regarding Batchnorm during training and testing.

Most of the post regarding older version where eg in Batchnorm there is a trainable paramter.

So, I wonder if now in the latest keras version I can use Batchnorm in training and in prediction (maybe using a generator or anything else) without taking care of any special things? 

Is there still an effect of the prediction batch size?

If using `validation_data `in model.fit, what does batchnorm take as statistics? Is it running averages from training?

```
model.fit(x,train_labels,
          batch_size=512,
          epochs=25,
          validation_data=(xtest, test_labels))

```
